src/functions/multi-transaction-to-webhook/handler.ts

```diff
@@ -0,0 +1,212 @@
+
+export const handler = async (event: CoreTransactionProcessedV2Event) => {
+  // get artifact url for the JSON version of EDI Transaction Set
+  const outputArtifactUrl = extractOutputArtifactUrl(event);
+
+  // get auth token to use when fetching output artifact
+  const { token } = await getAuthToken();
+
+  // fetch output artifact
+  const outputArtifactFetchResult = await fetch(outputArtifactUrl, {
+    method: "GET",
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  if (!outputArtifactFetchResult.ok) {
+    const artifactResponseSummary = {
+      statusText: outputArtifactFetchResult.statusText,
+      statusCode: outputArtifactFetchResult.status,
+      responseText: await outputArtifactFetchResult.text(),
+    };
+
+    throw new Error(
+      `Failed to retrieve output artifact: ${JSON.stringify(
+        artifactResponseSummary,
+        null,
+        2,
+      )}`,
+    );
+  }
+
+  // extract artifact JSON from fetch response body
+  const body = await outputArtifactFetchResult.json();
+
+  // load destination config for transaction set / partnership
+  const destinationConfig = await loadDestinationConfig(
+    event.detail.partnership.partnershipId,
+    event.detail.x12.metadata.transaction.transactionSetIdentifier,
+  );
+
+  // prepare string payload to send to destination webhook (with mapping optionally applied)
+  const webhookPayload =
+    destinationConfig.mappingId === undefined
+      ? JSON.stringify(body)
+      : await mappingResultAsString(destinationConfig, body);
+
+  // send stringified JSON to endpoint
+  const destinationWebhookFetchResult = await fetch(
+    destinationConfig.webhookUrl,
+    {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        ...(destinationConfig.authorization && {
+          Authorization: destinationConfig.authorization,
+        }),
+      },
+      body: webhookPayload,
+    },
+  );
+
+  const destinationWebhookResponseSummary = {
+    statusText: destinationWebhookFetchResult.statusText,
+    statusCode: destinationWebhookFetchResult.status,
+    responseText: await destinationWebhookFetchResult.text(),
+  };
+
+  if (!destinationWebhookFetchResult.ok) {
+    throw new Error(
+      `delivery to ${destinationConfig.webhookUrl} failed: ${JSON.stringify(
+        destinationWebhookResponseSummary,
+      )}`,
+    );
+  }
+
+  return destinationWebhookResponseSummary;
+};
+
+const extractOutputArtifactUrl = (
+  event: CoreTransactionProcessedV2Event,
+): string => {
+  const filteredOutputArtifacts = event.detail.artifacts.filter(
+    (artifact) => artifact.usage === "output",
+  );
+
+  if (filteredOutputArtifacts.length !== 1) {
+    throw new Error(
+      `Expected exactly 1 output artifact in event, received ${filteredOutputArtifacts.length}`,
+    );
+  }
+
+  const outputArtifactUrl = filteredOutputArtifacts[0]?.url;
+  if (!outputArtifactUrl) {
```

Comment:
Feels unnecessary, you checked the array length above, and url should always exist.

Reply:
@RossWilliams I completely agree with you, but without this check, TS still thinks this is `string | undefined`.... I even tried using a type predicate `.filter` above, but it still wasn't happy. if you have any wizardry here I would love to get rid of this check!

Reply:
updated this; resolving